src/lib/misc.cpp

```diff
@@ -838,3 +838,80 @@ userid_to_id(const uint8_t *userid, char *id)
     id[8 * 2] = 0x0;
     return id;
 }
+
+/* check whether string is hex */
+bool
+ishex(const char *hexid, size_t hexlen)
```

Comment:
Similar to the below, I think we have existing functionality for this.

Reply:
Ok but in the future I would say we should be moving towards a more C++-y approach.